### PR TITLE
[INT-292] Edit feature for Canva Integration

### PR DESCRIPTION
**Pull request by cmunozgar**
### Description
This PR adds the edit functionality for Canva within the composers. You will see a new edit button within the `MediaAttachmentThumbnail` component.

### Context & Notes
We want to add additional value to the customer by allowing them to edit an existing design in Canva. Any assets created through the integration will not be saved in a user’s folder in their Canva account, therefore we want to enable them to quickly edit a recently added image or a saved draft.

__Things to note__
We are adding a new attribute `source` to the `media` attribute within the `update` model. We will store the name of the source and the ID of the image.

We are changing the distribution of the action buttons (tag and edit) and put them in a separate container. The reason behind this is because the existing "tag" button is not really a button because it relies on the main button to be clicked. We'll leave the existing behavior of the onClick in the main button but specify ad-hoc onClick functions for these action buttons.

I've added the `box-sizing: border-box;` CSS rule to be pixel perfect with the current designs [here](https://www.figma.com/file/7HhG7w7DOcOkljxrbwW7Z7/Content-in-Integration?node-id=874%3A25328)

Places where the edit functionality is added:
- [X] MediaAttachmentThumbnail in the Composer
- [ ] CardItem in the StoryGroupComposer 

### Screenshots (if appropriate)
image

### Best Practices Checklist

-   [X] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [X] I have tested different user stories.
-   [X] I have identified similar or related features and tested they are still working.
-   [X] I have performed a self-review of my own code.
-   [X] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [X] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

### ⚠️ Pre-deploy Checklist

- [ ] I have manually loaded the URL of the staging deployment and confirmed it's working (see below**)
- [ ] I have confirmed that the Cypress E2E test is passing (_Note: we have a known issue where it doesn't run on some PRs. Hopefully this will be fixed soon._)
- [ ] I have confirmed that all other tests are passing.

** To visit the URL of the staging deployment, please vlick the link in the bot comment, or look under checks at the bottom of this PR and click "Details" next to `bufferbotbrains/cicd-buffer-publish`
image

### CODE REVIEW

File: packages/story-group-composer/middleware.js [hunk 1]
```diff
@@ -274,6 +274,7 @@ export default ({ getState, dispatch }) => next => action => {
             channelId: currentProfile.id,
             channelServiceId: currentProfile.serviceId,
             scheduledAt: formatShareDate(scheduledAt),
+            clientId: 'publishWeb',
```
Comment (anafilipadealmeida) [inline packages/story-group-composer/middleware.js:277]: Was this supposed to be `clientName` instead?

File: packages/composer/composer/components/MediaAttachmentThumbnail.jsx [hunk 2]
```diff
@@ -11,31 +12,40 @@ import Button from './shared/Button';
...
   background: rgba(0, 0, 0, 0.8);
   border-radius: ${borderRadius};
+  cursor: pointer;
+  margin-right: 4px;
 `;
 
-const renderTagOption = tagCount => (
-  <StyledTagWrapper>
+const renderTagOption = (tagCount, onClick) => (
+  <StyledTagWrapper onClick={onClick} data-tip="Tag">
```
Comment (anafilipadealmeida) [inline packages/composer/composer/components/MediaAttachmentThumbnail.jsx:34]: Semantically (specially for accessibility) it's not very good to have a div with a click handler and cursor pointer, behaving like a button, as screen readers and keyboards are not supported. If not using a button, I'd add a `role="button"` and `tabindex="0"` to it

File: packages/composer/composer/components/MediaAttachmentThumbnail.jsx [hunk 3]
```diff
@@ -11,31 +12,40 @@ import Button from './shared/Button';
...
     <StyledIconWrapper>
       <PersonIcon size="medium" />
     </StyledIconWrapper>
-    <Text>{tagCount || 'Tag'}</Text>
+    <Text>{tagCount}</Text>
+  </StyledTagWrapper>
+);
+
+const renderEditOption = onClick => (
+  <StyledTagWrapper data-tip="Edit"
```
Comment (anafilipadealmeida) [inline packages/composer/composer/components/MediaAttachmentThumbnail.jsx:43]: Same thing as the render tag option!

**Review by anafilipadealmeida — COMMENTED** [review]
😄

File: packages/story-group-composer/middleware.js (hunk 1 above)
Comment (cmunozgar) [reply packages/story-group-composer/middleware.js:277]: Good catch!

**Review by MayaUribe — APPROVED** [review]
Looks good @cmunozgar! Thank you for making the changes 🚀

**Pull request merged**